### ISSUE TITLE
[question] Implementing a custom graph node

### ISSUE THREAD

**Issue by akonneker** [alicevision/meshroom#1018]
I would like to develop a custom node which filters out blurry images by thresholding the variance of the Laplacian according to a user-defined threshold. Essentially this method: https://www.pyimagesearch.com/2015/09/07/blur-detection-with-opencv/.

So w.r.t. to this idea I have several questions: 

1. Is this sort of functionality something that would be considered for integration into the project? 
2. Which existing node should I look at for a simple of example of how to plumb everything?
3. Additionally, where should I look for info on how nodes are registered, etc. ?

Thanks for a great project!

**Comment by natowi** [comment]
The KeyframeSelection node has a blur detection option for videos.

For implementation details, take a look at https://github.com/natowi/meshroom_external_plugins/wiki/2-DevDoc

**Comment by cappelletto** [comment]
Hi @akonneker 
I implemented a couple of years ago a small C++/OpenCV module that extract frames from a video based on the spatial overlap and the image quality. As image quality I used something very similar to what you mention, by comparing the variance of the Laplacian between a set of candidate frames.
I was planning to convert it into a Meshroom node, so I am pretty interested to work towards the development of a node like the one you describe. However, I think I need to check KeyframeSelection thoroughly as it may already contain the desired functionality.

Cheers,
